plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java

```diff
@@ -71,11 +78,21 @@
 import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.extractSchema;
 import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnMappingMode;
 import static io.trino.plugin.hive.parquet.ParquetPageSourceFactory.PARQUET_ROW_INDEX_COLUMN;
+import static io.trino.spi.block.PageBuilderStatus.DEFAULT_MAX_PAGE_SIZE_IN_BYTES;
+import static java.lang.Math.min;
+import static java.lang.Math.toIntExact;
 import static java.util.Objects.requireNonNull;
 
 public class DeltaLakePageSourceProvider
         implements ConnectorPageSourceProvider
 {
+    // This is used whenever a query doesn't reference any data columns.
+    // We need to limit the number of rows per page in case there are projections
+    // in the query that can cause page sizes to explode. For example: SELECT rand() FROM some_table
+    // TODO (https://github.com/trinodb/trino/issues/16824) allow connector to return pages of arbitrary row count and handle this gracefully in engine
+    private static final int MAX_RLE_PAGE_SIZE = DEFAULT_MAX_PAGE_SIZE_IN_BYTES / LONG_INSTANCE_SIZE;
```

Comment:
I'm sorry i don't understand the division by `LONG_INSTANCE_SIZE`.
Same situation below - where I'd expect the division to happen by `LONG_INSTANCE_SIZE` only.

Could you please explain this choice?

Reply:
it actually should have been SIZE_OF_LONG.
RLE block has O(1) size, ignored
RLE page row count = DEFAULT_MAX_PAGE_SIZE_IN_BYTES (ideal page size) / the other column size.
Plus, we leave margin for "another other" column, just in case it gets projected. (of course, there could be _multiple_ additional columns projected, like in `SELECT rand(), rand(), rand()...` query, but we need to choose something)